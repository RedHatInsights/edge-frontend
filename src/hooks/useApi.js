import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/redux';
import { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';

const useApi = ({ api, id = null, statusMessages, tableReload = false }) => {
  const dispatch = useDispatch();
  const [data, setData] = useState([]);
  const [isLoading, setIsLoading] = useState(true);
  const [hasError, setHasError] = useState(false);
  const [apiFunc, setAPIFunc] = useState(() => api);

  const hasSuccess = statusMessages?.onSuccess;
  if (!statusMessages) {
    statusMessages = {
      onSuccess: {
        title: 'Success',
        description: 'The request has been made successfully',
      },
      onError: {
        title: 'Error',
        description: 'An error occurred making the request',
      },
    };
  }

  const fetchData = async (query = '') => {
    setIsLoading(true);
    try {
      const data = await apiFunc({ id, query });
      setData(data);

      if (hasSuccess) {
        dispatch({
          ...addNotification({
            variant: 'success',
            ...statusMessages.onSuccess,
          }),
        });
      }
    } catch (err) {
      setHasError(true);

      dispatch({
        ...addNotification({
          variant: 'danger',
          ...statusMessages.onError,
        }),
      });
    } finally {
      setIsLoading(false);
    }
  };

  useEffect(() => {
    if (!tableReload) {
      fetchData();
    }
  }, []);

  return [{ data, isLoading, hasError }, fetchData, apiFunc, setAPIFunc];
};

export default useApi;
