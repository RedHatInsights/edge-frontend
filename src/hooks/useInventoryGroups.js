import { useState, useEffect } from 'react';
import { useFeatureFlags } from '../utils';
import { FEATURE_PARITY_INVENTORY_GROUPS } from '../constants/features';
import { getEnforceEdgeGroups } from '../api/groups';

const useInventoryGroups = (value) => {
  const [data, setData] = useState(value);
  const inventoryGroupsEnabled = useFeatureFlags(
    FEATURE_PARITY_INVENTORY_GROUPS
  );

  useEffect(() => {
    (async () => {
      const response = await getEnforceEdgeGroups();
      const enforceEdgeGroups = response?.enforce_edge_groups;
      setData(!enforceEdgeGroups && inventoryGroupsEnabled);
    })();
  }, []);

  return data;
};

export default useInventoryGroups;
