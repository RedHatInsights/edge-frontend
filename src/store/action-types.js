import keyBy from 'lodash/keyBy';
import flatMap from 'lodash/flatMap';

export const LOAD_DEVICE_SUMMARY = 'LOAD_DEVICE_SUMMARY';
export const LOAD_ACTIVE_IMAGES = 'LOAD_ACTIVE_IMAGES';
export const LOAD_EDGE_IMAGES = 'LOAD_EDGE_IMAGES';
export const LOAD_EDGE_IMAGE_SETS = 'LOAD_EDGE_IMAGE_SETS';
export const LOAD_EDGE_IMAGE_PACKAGES = 'LOAD_EDGE_IMAGE_PACKAGES';
export const LOAD_GROUPS = 'LOAD_GROUPS';
export const LOAD_GROUP_DETAIL = 'LOAD_GROUP_DETAIL';
export const LOAD_TRESHOLD = 'LOAD_TRESHOLD';
export const LOAD_DEVICES_INFO = 'LOAD_DEVICES_INFO';
export const LOAD_CANARIES_INFO = 'LOAD_CANARIES_INFO';
export const LOAD_GROUP_DEVICES_INFO = 'LOAD_GROUP_DEVICES_INFO';
export const LOAD_IMAGE_STATUS = 'LOAD_IMAGE_STATUS';
export const LOAD_IMAGE_DETAIL = 'LOAD_IMAGE_DETAIL';
export const LOAD_IMAGE_SET_DETAIL = 'LOAD_IMAGE_SET_DETAIL';
export const CREATE_NEW_IMAGE = 'CREATE_NEW_IMAGE';
export const CREATE_NEW_IMAGE_RESET = `${CREATE_NEW_IMAGE}_RESET`;
export const POLLING_IMAGES = 'POLLING_IMAGES';
export const LOAD_DEVICE_TABLE = 'LOAD_DEVICE_TABLE';
export const UPDATE_DISPLAY_NAME = 'UPDATE_DISPLAY_NAME';
export const REMOVE_ENTITY = 'REMOVE_ENTITY';

const asyncActions = flatMap(
  [
    LOAD_EDGE_IMAGES,
    LOAD_EDGE_IMAGE_SETS,
    LOAD_EDGE_IMAGE_PACKAGES,
    LOAD_ACTIVE_IMAGES,
    LOAD_GROUPS,
    LOAD_GROUP_DETAIL,
    LOAD_TRESHOLD,
    LOAD_DEVICES_INFO,
    LOAD_CANARIES_INFO,
    LOAD_GROUP_DEVICES_INFO,
    LOAD_IMAGE_STATUS,
    LOAD_IMAGE_DETAIL,
    LOAD_IMAGE_SET_DETAIL,
    CREATE_NEW_IMAGE,
    LOAD_DEVICE_TABLE,
    UPDATE_DISPLAY_NAME,
    REMOVE_ENTITY,
  ],
  (a) => [a, `${a}_PENDING`, `${a}_FULFILLED`, `${a}_REJECTED`]
);
export const ACTION_TYPES = keyBy(asyncActions, (k) => k);

export const SELECT_ENTITY = 'SELECT_ENTITY';
export const PRE_SELECT_ENTITY = 'PRE_SELECT_ENTITY';
export const CLEAN_ENTITIES = 'CLEAN_ENTITIES';
