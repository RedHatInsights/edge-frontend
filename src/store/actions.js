import {
  ACTION_TYPES,
  LOAD_TRESHOLD,
  LOAD_DEVICES_INFO,
  LOAD_CANARIES_INFO,
  LOAD_GROUP_DETAIL,
  LOAD_GROUP_DEVICES_INFO,
  SELECT_ENTITY,
  PRE_SELECT_ENTITY,
  CLEAN_ENTITIES,
  LOAD_ACTIVE_IMAGES,
  LOAD_EDGE_IMAGES,
  LOAD_EDGE_IMAGE_SETS,
  LOAD_EDGE_IMAGE_PACKAGES,
  LOAD_DEVICE_SUMMARY,
  LOAD_IMAGE_STATUS,
  LOAD_IMAGE_DETAIL,
  LOAD_IMAGE_SET_DETAIL,
  CREATE_NEW_IMAGE,
  POLLING_IMAGES,
  LOAD_DEVICE_TABLE,
} from './action-types';

import {
  fetchImageStatus,
  getImageById,
  fetchEdgeImages,
  fetchEdgeImageSets,
  getImagePackageMetadata,
  createImage,
  getImageSet,
  fetchActiveImages,
} from '../api/images';
import { getInventory } from '../api/devices';
import { hosts } from '../api';
import { deleteSystemsById } from '../utils';

export const loadThreshold = () => ({
  type: LOAD_TRESHOLD,
  payload: () => {},
});

export const loadDevicesInfo = () => ({
  type: LOAD_DEVICES_INFO,
  payload: () => {},
});

export const loadCanariesInfo = () => ({
  type: LOAD_CANARIES_INFO,
  payload: () => {},
});

export const loadGroupsDetail = () => ({
  type: LOAD_GROUP_DETAIL,
  payload: () => {},
});

export const loadGroupDevicesInfo = () => ({
  type: LOAD_GROUP_DEVICES_INFO,
  payload: () => {},
});

export const selectEntity = (id, selected) => ({
  type: SELECT_ENTITY,
  payload: {
    id,
    selected,
  },
});

export const preSelectEntity = (id, selected) => ({
  type: PRE_SELECT_ENTITY,
  payload: {
    id,
    selected,
  },
});

export const cleanEntities = () => ({
  type: CLEAN_ENTITIES,
});

export const loadImages = (dispatch, pagination) => {
  dispatch({
    type: LOAD_ACTIVE_IMAGES,
    payload: fetchActiveImages(pagination),
    meta: {
      notifications: {
        rejected: {
          variant: 'danger',
          title: 'Cannot show images data',
          description: 'Failed receiving images from image-builder',
        },
      },
    },
    // the '.catch' part is necessary because redux-promise-middleware throws the error on REJECTED
    // and to avoid the app exploding I need to catch it here.
    // THANK you redux-promise-middleware for not allowing to customize this behavior. ðŸ˜ 
  }).catch(() => null);
};

export const loadDeviceSummary = (dispatch) => {
  dispatch({
    type: LOAD_DEVICE_SUMMARY,
    payload: () => {},
    meta: {
      notifications: {
        rejected: {
          variant: 'danger',
          title: 'Cannot show system summary data',
          description: 'Failed receiving system summary data from inventory',
        },
      },
    },
    // the '.catch' part is necessary because redux-promise-middleware throws the error on REJECTED
    // and to avoid the app exploding I need to catch it here.
    // THANK you redux-promise-middleware for not allowing to customize this behavior. ðŸ˜ 
  }).catch(() => null);
};

export const loadImageStatus = (dispatch, imageId) => {
  dispatch({
    type: LOAD_IMAGE_STATUS,
    payload: fetchImageStatus({ id: imageId }),
  }).catch(() => null);
};

export const loadImageDetail = (dispatch, imageId) => {
  dispatch({
    type: LOAD_IMAGE_DETAIL,
    payload: getImageById({ id: imageId }),
  }).catch(() => null);
};

export const createNewImage = (dispatch, payload, callback) => {
  dispatch({
    type: CREATE_NEW_IMAGE,
    payload: createImage(payload),
  })
    .then(callback)
    .catch(() => null);
};

export const loadEdgeImages = (dispatch, query) => {
  dispatch({
    type: LOAD_EDGE_IMAGES,
    payload: fetchEdgeImages(query),
  }).catch(() => null);
};

export const loadEdgeImageSets = (dispatch, query) => {
  dispatch({
    type: LOAD_EDGE_IMAGE_SETS,
    payload: fetchEdgeImageSets(query),
  }).catch(() => null);
};

export const setPolling = (toStart, interval) => {
  const subAction = toStart ? 'START' : 'END';
  const payload = toStart ? { interval } : {};
  return {
    type: `${POLLING_IMAGES}_${subAction}`,
    ...payload,
  };
};

export const addImageToPoll = ({ id, name }) => {
  return {
    type: `${POLLING_IMAGES}_ADD`,
    payload: {
      name,
      id,
    },
  };
};

export const removeImagesToPoll = (ids) => {
  return {
    type: `${POLLING_IMAGES}_REMOVE`,
    ids,
  };
};

export const loadImageSetDetail = (dispatch, urlParam, query) => {
  dispatch({
    type: LOAD_IMAGE_SET_DETAIL,
    payload: getImageSet({ id: urlParam, q: query }),
  }).catch(() => null);
};

export const loadImagePackageMetadata = (dispatch, imageId) => {
  dispatch({
    type: LOAD_EDGE_IMAGE_PACKAGES,
    payload: getImagePackageMetadata(imageId),
  }).catch(() => null);
};

export const loadDeviceTable = (dispatch) => {
  dispatch({
    type: LOAD_DEVICE_TABLE,
    payload: getInventory(),
  }).catch(() => null);
};

export const editDisplayName = (id, value, origValue) => ({
  type: ACTION_TYPES.UPDATE_DISPLAY_NAME,
  payload: hosts.apiHostPatchHostById([id], { display_name: value }),
  meta: {
    id,
    value,
    origValue,
    notifications: {
      fulfilled: {
        variant: 'success',
        title: `Display name for entity with ID ${id} has been changed to ${value}`,
        dismissable: true,
      },
    },
  },
});

export const deleteEntity = (systems, displayName) => ({
  type: ACTION_TYPES.REMOVE_ENTITY,
  payload: deleteSystemsById(systems),
  meta: {
    notifications: {
      fulfilled: {
        variant: 'success',
        title: 'Delete operation finished',
        description: `${displayName} has been successfully removed.`,
        dismissable: true,
      },
    },
    systems,
  },
});
