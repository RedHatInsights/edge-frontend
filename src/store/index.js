import { createContext } from 'react';
import { ReducerRegistry } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import promiseMiddleware from 'redux-promise-middleware';
import { notificationsMiddleware } from '@redhat-cloud-services/frontend-components-notifications/notificationsMiddleware';
import { polling } from './polling';

export const RegistryContext = createContext({
  getRegistry: () => {},
});

export function init(...middleware) {
  return new ReducerRegistry({}, [
    promiseMiddleware,
    notificationsMiddleware({
      errorDescriptionKey: ['detail', 'stack'],
    }),
    polling(),
    ...middleware,
  ]);
}
