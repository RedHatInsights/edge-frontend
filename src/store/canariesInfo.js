import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import { LOAD_CANARIES_INFO } from './action-types';

const initialState = {};

const loadCanariesInfoPending = (state) => ({
  ...state,
  canariesInfo: [],
  isLoading: true,
});
const loadCanariesInfoFulfilled = (state, { payload }) => ({
  ...state,
  canariesInfo: payload?.results || [],
  isLoading: false,
});

export default applyReducerHash(
  {
    [`${LOAD_CANARIES_INFO}_PENDING`]: loadCanariesInfoPending,
    [`${LOAD_CANARIES_INFO}_FULFILLED`]: loadCanariesInfoFulfilled,
  },
  initialState
);
