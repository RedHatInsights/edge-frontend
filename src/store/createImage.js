import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import { CREATE_NEW_IMAGE } from './action-types';

const initialState = {};

const createNewImagePending = () => ({
  isLoading: true,
  hasError: false,
  error: null,
});

const createNewImageFulfilled = () => {
  return {
    isLoading: false,
    hasError: false,
    error: null,
  };
};

const createNewImageRejected = (_state, { payload }) => ({
  isLoading: false,
  hasError: true,
  error: payload,
});

const createNewImageReset = () => ({
  isLoading: false,
  hasError: false,
  error: null,
});

export default applyReducerHash(
  {
    [`${CREATE_NEW_IMAGE}_PENDING`]: createNewImagePending,
    [`${CREATE_NEW_IMAGE}_FULFILLED`]: createNewImageFulfilled,
    [`${CREATE_NEW_IMAGE}_REJECTED`]: createNewImageRejected,
    [`${CREATE_NEW_IMAGE}_RESET`]: createNewImageReset,
  },
  initialState
);
