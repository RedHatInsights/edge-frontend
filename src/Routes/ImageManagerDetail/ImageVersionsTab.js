import React, { useEffect, useState } from 'react';
import GeneralTable from '../../components/general-table/GeneralTable';
import PropTypes from 'prop-types';
import { routes as paths } from '../../../package.json';
import { Link } from 'react-router-dom';
import { Text } from '@patternfly/react-core';
import { DateFormat } from '@redhat-cloud-services/frontend-components/DateFormat';
import StatusLabel from '../ImageManagerDetail/StatusLabel';
import { imageTypeMapper } from '../ImageManagerDetail/constants';
import { loadImageSetDetail } from '../../store/actions';
import { cellWidth } from '@patternfly/react-table';
import Main from '@redhat-cloud-services/frontend-components/Main';

const defaultFilters = [
  {
    label: 'Status',
    type: 'checkbox',
    options: [
      { option: 'Building', value: 'BUILDING' },
      { option: 'Created', value: 'CREATED' },
      { option: 'Error', value: 'ERROR' },
      { option: 'Ready', value: 'SUCCESS' },
    ],
  },
];

const columnNames = [
  {
    title: 'Version',
    type: 'version',
    sort: true,
    columnTransforms: [cellWidth(15)],
  },
  {
    title: 'Output',
    type: 'image_type',
    sort: false,
    columnTransforms: [cellWidth(35)],
  },
  {
    title: 'Created',
    type: 'created_at',
    sort: true,
    columnTransforms: [cellWidth(25)],
  },
  {
    title: 'Status',
    type: 'status',
    sort: false,
    columnTransforms: [cellWidth(35)],
  },
];

const createRows = (data, imageSetId) => {
  return data?.map(({ image }) => ({
    rowInfo: {
      id: image?.ID,
      imageStatus: image?.Status,
      isoURL: image?.Installer?.ImageBuildISOURL,
    },
    noApiSortFilter: [
      image?.Version,
      imageTypeMapper[image?.ImageType],
      image?.CreatedAt,
      image?.Status,
    ],
    cells: [
      {
        title: (
          <Link
            to={`${paths['manage-images']}/${imageSetId}/versions/${image.ID}/details`}
          >
            {image?.Version}
          </Link>
        ),
      },
      {
        title: imageTypeMapper[image?.ImageType],
      },
      {
        title: <DateFormat date={image?.CreatedAt} />,
      },
      {
        title: <StatusLabel status={image?.Status} />,
      },
    ],
  }));
};

const ImageVersionsTab = ({ imageData, openUpdateWizard }) => {
  const [rows, setRows] = useState([]);
  useEffect(() => {
    if (imageData?.data) {
      setRows(
        createRows(
          imageData?.data?.Data?.images,
          imageData?.data?.Data?.image_set?.ID
        )
      );
    }
  }, [imageData]);

  const actionResolver = (rowData) => {
    const actionsArray = [];
    if (rowData.rowInfo?.isoURL) {
      actionsArray.push({
        title: (
          <Text
            className="force-text-black remove-underline"
            component="a"
            href={rowData.rowInfo.isoURL}
            rel="noopener noreferrer"
            target="_blank"
          >
            Download
          </Text>
        ),
      });
    }
    let imageLatestVersion = imageData?.data?.Data?.images?.[0].image?.Version;

    const areActionsDisabledForNonLatestImageVersion = (imageData) => {
      for (let i = 0; i < imageLatestVersion; i++) {
        console.log(imageData?.data?.Data?.images?.[i].image?.Version);
        if (
          imageData?.data?.Data?.images?.[i].image?.Version < imageLatestVersion
        ) {
          return true;
        }
      }
    };
    if (
      rowData.rowInfo?.imageStatus === 'SUCCESS' ||
      rowData.rowInfo?.imageStatus === 'ERROR'
    ) {
      actionsArray.push({
        title: 'Update Image',
        isDisabled: areActionsDisabledForNonLatestImageVersion,
        onClick: (_event, _rowId, rowData) => {
          openUpdateWizard(rowData.rowInfo.id);
        },
      });
    }

    if (rowData.rowInfo?.imageStatus === 'BUILDING' && rowData.rowInfo?.id) {
      actionsArray.push({
        title: '',
      });
    }
    return actionsArray;
  };

  const areActionsDisabled = (rowData) =>
    rowData.rowInfo?.imageStatus === 'BUILDING';

  return (
    <Main className="add-100vh">
      <GeneralTable
        apiFilterSort={false}
        filters={defaultFilters}
        loadTableData={loadImageSetDetail}
        tableData={{
          count: imageData?.data?.Count,
          isLoading: imageData?.isLoading,
          hasError: imageData?.hasError,
        }}
        columnNames={columnNames}
        rows={rows || []}
        actionResolver={actionResolver}
        areActionsDisabled={areActionsDisabled}
        defaultSort={{ index: 2, direction: 'desc' }}
      />
    </Main>
  );
};
ImageVersionsTab.propTypes = {
  imageData: PropTypes.object,
  urlParam: PropTypes.string,
  openUpdateWizard: PropTypes.func,
};

export default ImageVersionsTab;
