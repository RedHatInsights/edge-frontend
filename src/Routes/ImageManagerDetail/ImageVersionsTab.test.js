import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import ImageVersionsTab from './ImageVersionsTab';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import logger from 'redux-logger';
import { init, RegistryContext } from '../../store';
import * as useApi from '../../hooks/useApi';

describe('ImageVersionsTab', () => {
  const imagesViewData = {
    count: 2,
    data: [
      {
        ID: 200,
        Version: 2,
        ImageType: 'rhel-edge-installer',
        CreatedAt: '2022-07-15T16:41:44.237455+02:00',
        Status: 'BUILDING',
        ImageBuildIsoURL: '',
      },
      {
        ID: 100,
        Version: 1,
        ImageType: 'rhel-edge-installer',
        CreatedAt: '2022-07-15T13:52:09.155502+02:00',
        Status: 'SUCCESS',
        ImageBuildIsoURL: 'test.com',
      },
    ],
  };

  const mockFetchData = jest.fn();

  beforeEach(() => {
    jest
      .spyOn(useApi, 'default')
      .mockImplementation(() => [
        { data: imagesViewData, isLoading: false, hasError: false },
        mockFetchData,
      ]);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('renders correctly', async () => {
    const openUpdateWizard = jest.fn();
    const imageData = {
      data: {
        ImageBuildIsoURL: 'test.com',
        ImageSet: {
          ID: 101,
        },
        LastImageDetails: {
          image: {
            ID: 200,
            Status: 'BUILDING',
            Version: 2,
            ImageType: 'rhel-edge-installer',
          },
        },
      },
    };

    const registry = init(logger);

    const { container } = render(
      <RegistryContext.Provider
        value={{
          getRegistry: () => registry,
        }}
      >
        <Provider store={registry.getStore()}>
          <ImageVersionsTab
            imageData={imageData}
            openUpdateWizard={openUpdateWizard}
          />
        </Provider>
      </RegistryContext.Provider>,
      { wrapper: MemoryRouter }
    );

    expect(mockFetchData.mock.calls.length).toBe(1);

    expect(screen.getByRole('button', { name: 'Options menu' })).toBeDefined();
    expect(
      screen
        .getByRole('cell', { name: '1' })
        .firstChild.getAttribute('target-href')
    ).toEqual('edge/manage-images/101/versions/100/details');
    expect(
      screen
        .getByRole('cell', { name: '2' })
        .firstChild.getAttribute('target-href')
    ).toEqual('edge/manage-images/101/versions/200/details');
    expect(
      screen.getAllByRole('cell', { name: 'RHEL for Edge Installer (.iso)' })
    ).toHaveLength(2);
    expect(screen.getAllByText(/ago/i)).toBeDefined();
    expect(
      screen.getByRole('cell', { name: 'Image build in progress' })
    ).toBeDefined();
    expect(screen.getByRole('cell', { name: 'Ready' })).toBeDefined();
    fireEvent.click(
      screen.getAllByRole('button', { name: /Kebab toggle/i })[1]
    );
    // fireEvent.click(screen.getByRole('button', { name: /update image/i }));
    // expect(openUpdateWizard).toBeCalled();

    expect(container.querySelector('div')).toMatchSnapshot();
  });
});
