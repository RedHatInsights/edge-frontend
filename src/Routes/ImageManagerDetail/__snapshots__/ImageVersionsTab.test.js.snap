// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageVersionsTab renders correctly 1`] = `
<div
  class="pf-c-toolbar"
  data-ouia-component-id="OUIA-Generated-Toolbar-1"
  data-ouia-component-type="PF4/Toolbar"
  data-ouia-safe="true"
  data-testid="toolbar-header-testid"
  id="toolbar-header"
  style="padding: 0px;"
>
  <div
    class="pf-c-toolbar__content"
  >
    <div
      class="pf-c-toolbar__content-section"
    >
      <div
        class="pf-c-toolbar__item"
        data-testid="filter-input-testid"
      >
        <div
          class="pf-c-input-group"
        >
          <div
            class="pf-c-select"
            data-ouia-component-id="OUIA-Generated-Select-checkbox-1"
            data-ouia-component-type="PF4/Select"
            data-ouia-safe="true"
            style="width: 11rem;"
          >
            <button
              aria-expanded="false"
              aria-label="Options menu"
              aria-labelledby=" pf-select-toggle-id-0"
              class="pf-c-select__toggle"
              id="pf-select-toggle-id-0"
              type="button"
            >
              <div
                class="pf-c-select__toggle-wrapper"
              >
                <span
                  class="pf-c-select__toggle-text"
                >
                  Filter by status
                </span>
              </div>
              <span
                class="pf-c-select__toggle-arrow"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 320 512"
                  width="1em"
                >
                  <path
                    d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        align="[object Object]"
        class="pf-c-toolbar__item pf-m-pagination"
      >
        <div
          class="pf-c-pagination pf-m-compact"
          data-ouia-component-id="OUIA-Generated-Pagination-top-1"
          data-ouia-component-type="PF4/Pagination"
          data-ouia-safe="true"
          data-testid="pagination-header-test-id"
          id="pagination-options-menu-top-top-pagination"
          style="--pf-c-pagination__nav-page-select--c-form-control--width-chars: 2;"
        >
          <div
            class="pf-c-pagination__total-items"
          >
            <b>
              1
               - 
              2
            </b>
             
            of
             
            <b>
              2
            </b>
             
            
          </div>
          <div
            class="pf-c-options-menu"
            data-ouia-component-type="PF4/PaginationOptionsMenu"
            data-ouia-safe="true"
          >
            <div
              class="pf-c-options-menu__toggle pf-m-plain pf-m-text"
            >
              <span
                class="pf-c-options-menu__toggle-text"
              >
                <b>
                  1
                   - 
                  2
                </b>
                 
                of
                 
                <b>
                  2
                </b>
                 
                
              </span>
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-label="Items per page"
                class="  pf-c-options-menu__toggle-button"
                data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                data-ouia-component-type="PF4/DropdownToggle"
                data-ouia-safe="true"
                id="pagination-options-menu-top-top-toggle"
                type="button"
              >
                <span
                  class="pf-c-options-menu__toggle-button-icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 320 512"
                    width="1em"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
          <nav
            aria-label="Pagination"
            class="pf-c-pagination__nav"
          >
            <div
              class="pf-c-pagination__nav-control"
            >
              <button
                aria-disabled="true"
                aria-label="Go to previous page"
                class="pf-c-button pf-m-plain pf-m-disabled"
                data-action="previous"
                data-ouia-component-id="OUIA-Generated-Button-plain-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                disabled=""
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                  />
                </svg>
              </button>
            </div>
            <div
              class="pf-c-pagination__nav-control"
            >
              <button
                aria-disabled="true"
                aria-label="Go to next page"
                class="pf-c-button pf-m-plain pf-m-disabled"
                data-action="next"
                data-ouia-component-id="OUIA-Generated-Button-plain-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                disabled=""
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                  />
                </svg>
              </button>
            </div>
          </nav>
        </div>
      </div>
    </div>
    <div
      class="pf-c-toolbar__expandable-content"
      id="toolbar-header-expandable-content-0"
    >
      <div
        class="pf-c-toolbar__group"
      />
    </div>
  </div>
  <div
    class="pf-c-toolbar__content"
  >
    <div
      class="pf-c-toolbar__content-section"
    >
      <div
        class="pf-c-toolbar__item pf-m-chip-group pf-m-spacer-none"
      />
    </div>
    <div
      class="pf-c-toolbar__expandable-content"
      id="toolbar-header-expandable-content-1"
    >
      <div
        class="pf-c-toolbar__group"
      />
    </div>
  </div>
  <div
    class="pf-c-toolbar__content pf-m-hidden"
    hidden=""
  >
    <div
      class="pf-c-toolbar__group"
    />
  </div>
</div>
`;
